Source/WebCore/page/Page.cpp

```diff
@@ -5086,4 +5087,75 @@ bool Page::requiresScriptTelemetryForURL(const URL& scriptURL) const
     return chrome().client().requiresScriptTelemetryForURL(scriptURL, mainFrameOrigin());
 }
 
+void Page::applyWindowFeatures(const WindowFeatures& features)
+{
+    Ref frame = mainFrame();
+    chrome().setToolbarsVisible(features.toolBarVisible || features.locationBarVisible);
+
+    if (!frame->page())
+        return;
+    if (features.statusBarVisible)
+        chrome().setStatusbarVisible(*features.statusBarVisible);
+
+    if (!frame->page())
+        return;
+    if (features.scrollbarsVisible)
+        chrome().setScrollbarsVisible(*features.scrollbarsVisible);
+
+    if (!frame->page())
+        return;
+    if (features.menuBarVisible)
+        chrome().setMenubarVisible(*features.menuBarVisible);
+
+    if (!frame->page())
+        return;
```

Comment:
Doesn’t seem like all these null checks are still needed since we are accessing `chrome()` from `this` now. Callers should have a RefPtr on the stack though.

Reply:
Many of them were probably never needed, but I kept them all to minimize change in behavior.